server/user/activity-tracker.js

```diff
@@ -0,0 +1,48 @@
+'use strict';
+
+const { User } = require('../common/database');
+const debounce = require('lodash/debounce');
+const throttle = require('lodash/throttle');
+
+const toMiliseconds = min => min * 60000;
+
+class ActivityTracker {
+  constructor({ saveInterval, ttl }) {
+    this.tracked = {};
+    this.saveInterval = toMiliseconds(saveInterval);
+    this.ttl = toMiliseconds(ttl);
+  }
+
+  track(id) {
+    if (!this.tracked[id]) {
+      this.tracked[id] = {
+        save: throttle(() => this.save(id), this.saveInterval),
+        untrack: debounce(() => this.untrack(id), this.ttl)
+      };
+    }
+    this.tracked[id].lastActive = new Date();
+    this.tracked[id].untrack();
```

Comment:
If I read this correctly you used `debounce` to mimic `setInterval` behavior? 🙃 You are scheduling actual `untrack` to be execute after `ttl` elapses through `_.debounce` and afterwards canceling debounce _queue_. How's that different than simply using:
```js
setInterval(() => this.tracked[id].untrack(), this.ttl);
```
😕